core-tests/src/test/java/org/mule/el/mvel/datatype/PropertyEnricherDataTypePropagatorTestCase.java

```diff
@@ -37,17 +34,24 @@
     @Test
     public void propagatesDataTypeForInlinedInvocationProperty() throws Exception
     {
-        doInlinePropertyDataTypePropagationTest(INVOCATION);
+        final DataType expectedDataType = DataTypeFactory.create(String.class, JSON);
```

Comment:
In all of these session/flowVar test, looks to me that the only different part is how the test reads/writes data. Can these test be refactored to avoid the duplication?